ansible/create-vms.yml

```diff
@@ -0,0 +1,84 @@
+--- 
+
+- name: Create VMs
+  vars_files:
+    - vars/all.yml
+  hosts: localhost
+  strategy: free
+  tasks:
+    - name: Create VMs From Family
+      include_role:
+        name: create-vm
+      vars:
+        vm:
+          name: "{{ gcp_instance_prefix }}-{{ item.1 }}-{{ job_id }}"
+          project: "{{ item.0.project }}"
+          family: "{{ item.1 }}"
+          labels:
+            # labels can be used to pass useful information to the dynamic
+            # inventory (which is populated after the VMs are created)
+            job_id: "{{ job_id }}"
+            platform: "{{ item.0.name }}"
+            vm_config: "{{ item.0.name }}_{{ item.1 }}"
+            username: "{{ item.0.username | default(gcp_default_user) }}"
+
+      #
+      # The loop will create the "item" variable, which can be used throughout the task
+      # and the subelements call will create a list that combines the project
+      # with all of the image families in that set
+      # e.g.
+      #   - name: rhel
+      #     families:
+      #      - rhel-7
+      #      - rhel-8
+      #
+      #  becomes:
+      #  [{'name': 'rhel', 'families': ['rhel-7', 'rhel-8']}, 'rhel-7']
+      #  [{'name': 'rhel', 'families': ['rhel-7', 'rhel-8']}, 'rhel-8']
+      #
+      #  This also allows us to filter based on specific test_platforms as seen
+      #  in the when clause
+      #
+      loop: "{{ virtual_machines | subelements('families', skip_missing=True) }}"
+      when: "test_platform == item.0.name or test_platform == 'all'"
+
+    - name: Create VMs From Image
+      include_role:
+        name: create-vm
+      vars:
+        vm:
```

Comment:
I wanted a dictionary to create something of a 'namespace' for the variables for the role (so you can do something like `vm.labels` instead of just `labels` or `vm_labels`

Reply:
I ended up turning this into a normal list of variables after refactoring most of the labelling to the VM creation role instead. So now everything is `vm_` prefixed and feels a fair bit neater